File: model/src/main/java/no/nav/pensjon/vtp/testmodell/identer/IdenterIndeks.java
package no.nav.pensjon.vtp.testmodell.identer;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import org.springframework.stereotype.Component;

@Component
public class IdenterIndeks {
    private final Map<String, LokalIdentIndeks> identer = new ConcurrentHashMap<>();
    private final IdentGenerator identGenerator = new FiktiveFnr();

    public LokalIdentIndeks getIdenter(String unikScenarioId) {
        return identer.computeIfAbsent(unikScenarioId, n -> new LokalIdentIndeks(n, identGenerator));
    }
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/identer/IdenterIndeks.kt
package no.nav.pensjon.vtp.testmodell.identer

import org.springframework.stereotype.Component
import java.util.concurrent.ConcurrentHashMap

@Component
class IdenterIndeks {
    private val identer: MutableMap<String, LokalIdentIndeks> = ConcurrentHashMap()
    private val identGenerator: IdentGenerator = FiktiveFnr()

    fun getIdenter(unikScenarioId: String): LokalIdentIndeks {
        return identer.computeIfAbsent(unikScenarioId) { LokalIdentIndeks(it, identGenerator) }
    }
}
